### PR TITLE
kafka(ticdc): a new kafka admin client implementation based on kafka-go

### LINKED ISSUE #8092: implement kafka-go based admin client

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8092

### What is changed and how it works?

* a new admin client implementation based on the kafka-go library.
* use gomock to mock kafka-go client, to support unit test.

* unit test coverage is around 96%

Screenshot 2023-01-30 at 15 58 58

#### Tests 

 - Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/kafka/v2/client.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+// Client is the interface of the kafka-go client, it contains a subset of all methods which is used
+// by the kafka sink.
+// This interface is mainly used to support mock kafka-go client in unit test.
+type Client interface {
+	Metadata(ctx context.Context, req *kafka.MetadataRequest) (*kafka.MetadataResponse, error)
+	DescribeConfigs(ctx context.Context,
+		req *kafka.DescribeConfigsRequest) (*kafka.DescribeConfigsResponse, error)
+	CreateTopics(ctx context.Context,
+		req *kafka.CreateTopicsRequest) (*kafka.CreateTopicsResponse, error)
```
Comment (overvenus) [inline pkg/sink/kafka/v2/client.go:30]:
```suggestion
	DescribeConfigs(
		ctx context.Context, req *kafka.DescribeConfigsRequest,
	) (*kafka.DescribeConfigsResponse, error)
	CreateTopics(
		ctx context.Context, req *kafka.CreateTopicsRequest,
	) (*kafka.CreateTopicsResponse, error)
```

File: cdc/sinkv2/metrics/mq/kafka/collector.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ func (m *Collector) Run(ctx context.Context) {
 
 func (m *Collector) updateBrokers() {
```
Comment (overvenus) [inline cdc/sinkv2/metrics/mq/kafka/collector.go:92]:
```suggestion
func (m *Collector) updateBrokers(ctx context.Context) {
```

There is a context in the caller of `updateBrokers`.

File: cdc/sink/mq/producer/kafka/metrics.go [hunk 3]
```diff
@@ -228,7 +228,7 @@ func getBrokerMetricName(prefix, brokerID string) string {
 
 func (sm *saramaMetricsMonitor) collectBrokers() {
```
Comment (overvenus) [inline cdc/sink/mq/producer/kafka/metrics.go:229]:
```suggestion
func (sm *saramaMetricsMonitor) collectBrokers(ctx context.Context) {
```

File: pkg/sink/kafka/admin.go [hunk 4]
```diff
@@ -58,15 +58,15 @@ func (a *saramaAdminClient) GetAllBrokers() ([]Broker, error) {
 	return result, nil
 }
 
-func (a *saramaAdminClient) GetCoordinator() (int32, error) {
+func (a *saramaAdminClient) GetCoordinator(context.Context) (int, error) {
```
Comment (sdojjy) [inline pkg/sink/kafka/admin.go:61]: why change int32 to int?

Comment (3AceShowHand) [reply pkg/sink/kafka/admin.go:61]: because it's more easy to handle in the kafka-go implementation.

File: pkg/sink/kafka/v2/admin.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	"github.com/segmentio/kafka-go"
+	"go.uber.org/zap"
+)
+
+type admin struct {
+	client Client
+}
+
+// NewClusterAdminClient return an admin client
+func NewClusterAdminClient(endpoints []string) *admin {
```
Comment (sdojjy) [inline pkg/sink/kafka/v2/admin.go:32]: change the return type to  `ClusterAdminClient`

**Review by overvenus — APPROVED** [review]

File: pkg/sink/kafka/v2/admin.go [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+				ResourceType: kafka.ResourceTypeBroker,
+				ResourceName: strconv.Itoa(controllerID),
+				ConfigNames:  []string{configName},
+			},
+		},
+	}
+
+	resp, err := a.client.DescribeConfigs(ctx, request)
+	if err != nil {
+		return "", err
```
Comment (sdojjy) [inline pkg/sink/kafka/v2/admin.go:105]:
```suggestion
		return "", errors.Trace(err)
```

File: pkg/sink/kafka/v2/admin.go [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+		describeTopicConfigsRequest.Resources = append(describeTopicConfigsRequest.Resources,
+			kafka.DescribeConfigRequestResource{
+				ResourceType: kafka.ResourceTypeTopic,
+				ResourceName: topic.Name,
+			})
+	}
+
+	describeTopicConfigsResponse, err := a.client.DescribeConfigs(ctx, describeTopicConfigsRequest)
+	if err != nil {
+		return nil, err
```
Comment (sdojjy) [inline pkg/sink/kafka/v2/admin.go:151]:
```suggestion
		return nil, errors.Trace(err)
```

**Review by sdojjy — APPROVED** [review]

**Pull request merged**